spec.md

```diff
@@ -133,7 +133,44 @@ If the blob is not found in the registry, the response code MUST be `404 Not Fou
 
 #### Push
 Pushing an image works in the opposite order as a pull: the blobs making up the layers are pushed first, and the
-manifest last.
+manifest last. A useful diagram is provided [here](https://github.com/google/go-containerregistry/tree/d7f8d06c87ed209507dd5f2d723267fe35b38a9f/pkg/v1/remote#anatomy-of-an-image-upload).
```

Comment:
If we don't want to just link out to another repo but find it useful, the source for this is here: https://github.com/google/go-containerregistry/blob/master/images/dot/upload.dot

Reply:
I think it would be redundant to re-host it, and what if the other version changes but ours does not? What do others think?